sql/src/test/java/io/crate/integrationtests/UnionIntegrationTest.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.integrationtests;
+
+import io.crate.action.sql.SQLActionException;
+import io.crate.testing.UseJdbc;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.arrayContaining;
+import static org.hamcrest.Matchers.arrayContainingInAnyOrder;
+import static org.hamcrest.Matchers.is;
+
+@ESIntegTestCase.ClusterScope(minNumDataNodes = 2, maxNumDataNodes = 5)
+@UseJdbc
+public class UnionIntegrationTest extends SQLTransportIntegrationTest {
+
+    private boolean setupDone;
+
+    @Before
+    public void beforeTest() {
+        if (!setupDone) {
```

Comment:
I think this doesn't work as you'd expect it to? setupDone is re-initialized per test. I'd have to be static and then you've to be careful with synchronization.

Reply:
Wasn't aware that JUnit creates a new instance of the class for every test.